light-model-client/src/jvmTest/kotlin/org/modelix/client/light/FakeLightModelServer.kt

```diff
@@ -79,142 +78,138 @@ class FakeLightModelServer(val client: LocalModelClient, val repositoriesManager
+                                            .asSequence()
+                                            .filterIsInstance<AddNewChildNodeOpData>()
+                                            .map { it.childId }
+                                            .filter { previouslyReplacedIds[it] == null }
+                                            .forEach { replacedIds[it] = repositoriesManager.stores.idGenerator.generate().toString(16) }
+                                        val idsKnownByClient = HashSet<String>()
+                                        val operations = message.operations!!
+                                            .map {
+                                                it.replaceIds { id ->
+                                                    idsKnownByClient.add(id)
+                                                    previouslyReplacedIds[id] ?: replacedIds[id]
                                                 }
-                                            val baseVersion = if (message.baseChangeSet == null) {
-                                                lastVersionToClient
-                                            } else {
-                                                lastVersionFromClient ?: lastVersionToClient
                                             }
-
-                                            val mergedVersion = applyUpdate(baseVersion!!, operations, repositoryId, userId)
-                                            lastVersionFromClient = mergedVersion
-                                            sendDelta(mergedVersion, replacedIds, message.changeSetId)
-                                            previouslyReplacedIds.putAll(replacedIds)
-                                            // idsKnownByClient.forEach { previouslyReplacedIds.remove(it) }
-                                            // TODO remove all other entries in `previouslyReplacedIds` that were part of the same message
+                                        val baseVersion = if (message.baseChangeSet == null) {
+                                            lastVersionToClient
+                                        } else {
+                                            lastVersionFromClient ?: lastVersionToClient
                                         }
+
+                                        val mergedVersion = applyUpdate(baseVersion!!, operations, repositoryId, userId)
+                                        lastVersionFromClient = mergedVersion
+                                        sendDelta(mergedVersion, replacedIds, message.changeSetId)
+                                        previouslyReplacedIds.putAll(replacedIds)
+                                        // idsKnownByClient.forEach { previouslyReplacedIds.remove(it) }
+                                        // TODO remove all other entries in `previouslyReplacedIds` that were part of the same message
```

Comment:
Is this TODO still relevant?

Reply:
The diff doesn't show it very well, but I only removed the wrapping `repositoriesManager.runWithRepository(repositoryId) { ... }`. I will not change anything else, at least not in this PR.